cms/djangoapps/contentstore/views/component.py

```diff
@@ -308,8 +308,9 @@ def create_support_legend_dict():
                             )
                         )
 
-        # Add any advanced problem types. Note that these are different xblocks being stored as Advanced Problems.
-        if category == 'problem':
+        # Add any advanced problem types. Note that these are different xblocks being stored as Advanced Problems,
+        # currently not supported in libraries .
+        if category == 'problem' and not library:
```

Comment:
Thanks for picking this up! The code looks good, but please add a test that these advanced modules do not show up for libraries (and make sure that the test fails without your fix).

Note that TestComponentTemplates in test_item.py shows how to properly set everything up so that 'openassessment' and 'drag-and-drop-v2' show up for non-libraries. However, I think the best place to put the new test case is within test_get_component_templates in test_library.py.

````
def test_get_component_templates(self):
        """
        Verify that templates for adding discussion and advanced components to
        content libraries are not provided.
        """
        lib = LibraryFactory.create()
        lib.advanced_modules = ['lti']
        lib.save()
        templates = [template['type'] for template in get_component_templates(lib, library=True)]
        self.assertIn('problem', templates)
        self.assertNotIn('discussion', templates)
        self.assertNotIn('advanced', templates)